### ISSUE TITLE
Deep Convolutional Generative Adversarial Networks training question

### ISSUE THREAD

**Issue by kli-nlpr** [zackchase/mxnet-the-straight-dope#526]
In
```
with autograd.record():
            # train with real image
            output = netD(data).reshape((-1, 1))
            errD_real = loss(output, real_label)
            metric.update([real_label,], [output,])

            # train with fake image
            fake = netG(latent_z)
            output = netD(fake.detach()).reshape((-1, 1))
            errD_fake = loss(output, fake_label)
            errD = errD_real + errD_fake
            errD.backward()
            metric.update([fake_label,], [output,])
```
I am confused by this line of code
```
fake = netG(latent_z)
```
In my opinion, this line is used to generate fake data, and `netG` should in `predict mode`, am I right?
If I am right, since `fake = netG(latent_z)` is in the context of `with autograd.record():`, which imply that `netG` is in `train mode` .  (https://gluon.mxnet.io/chapter03_deep-neural-networks/mlp-dropout-gluon.html#Integration-with-autograd.record)
